src/inventory/ModelManager.cpp

```diff
@@ -741,46 +759,114 @@ QVariant ModelManager::headerData(int section, Qt::Orientation orientation, int
+            return tr("Local Version", "Version of the locally installed model.");
+        case Column::RemoteVer:
+            return tr("Remote Version", "Version of the remote model.");
+        case Column::Installed:
+            return tr("Installed", "Whether the model is available locally, remotely or is out of date.");
         default:
             return QVariant();
     }
 }
 
 QVariant ModelManager::data(const QModelIndex &index, int role) const {
-    if (index.row() >= localModels_.size())
+    if (index.row() >= localModels_.size() + newModels_.size())
         return QVariant();
 
-    Model model = localModels_[index.row()];
+    Model model; // Make sure we have all local models before the remote ones
+    if (index.row() < localModels_.size())
+        model = localModels_[index.row()];
+    else if (index.row() < localModels_.size() + newModels_.size())
+        model = newModels_[index.row() - localModels_.size()];
 
     if (role == Qt::UserRole)
         return QVariant::fromValue(model);
 
     switch (index.column()) {
-        case Column::Name:
+        case Column::Source:
             switch (role) {
                 case Qt::DisplayRole:
-                    return model.modelName;
+                    return model.src;
+                case Qt::BackgroundRole:
+                    if (model.isLocal()) {
+                        return QColor(0xD8, 0xF1, 0xBF);
+                    } else {
+                        return QColor(0xE7, 0xE8, 0xE6);
+                    }
+                case Qt::ForegroundRole:
+                    if (model.outdated()) {
+                        return QColor(0xFD, 0x25, 0x58);
+                    }
                 default:
                     return QVariant();
             }
 
+        case Column::Target:
+            switch (role) {
+                case Qt::DisplayRole:
+                    return model.trg;
+                case Qt::BackgroundRole:
+                    if (model.isLocal()) {
+                        return QColor(0xD8, 0xF1, 0xBF);
+                    } else {
+                        return QColor(0xE7, 0xE8, 0xE6);
+                    }
+                case Qt::ForegroundRole:
+                    if (model.outdated()) {
+                        return QColor(0xFD, 0x25, 0x58);
+                    }
+                default:
+                    return QVariant();
+            }
+
+        case Column::Type:
+        switch (role) {
+            case Qt::DisplayRole:
+                return model.type;
+            case Qt::BackgroundRole:
+                if (model.isLocal()) {
+                    return QColor(0xD8, 0xF1, 0xBF);
+                } else {
+                    return QColor(0xE7, 0xE8, 0xE6);
+                }
+            case Qt::ForegroundRole:
+                if (model.outdated()) {
+                    return QColor(0xFD, 0x25, 0x58);
+                }
+            default:
+                return QVariant();
+        }
```

Comment:
The indent is different here from the other switch cases (Sorry I hate leaving these comments :( )